diff --git a/tools/copy-to-sha256.py b/tools/copy-to-sha256.py
index 96e2b9b6..c56a859f 100755
--- a/tools/copy-to-sha256.py
+++ b/tools/copy-to-sha256.py
@@ -75,8 +75,17 @@ def handle_dir(logger, from_path: str, to_path: str):
 
 def handle_tar(logger, tar, to_path: str):
     for member in tar.getmembers():
-        if member.isfile() or member.islnk():
-            f = tar.extractfile(member)
+        logging.info(f'Checking {member.name}')
+
+        # Manually resolve hardlinks first
+        resolved_member = member
+        while resolved_member.islnk():
+            # Tar hardlink members' linknames are already relative to root of the archive
+            resolved_member = tar.getmember(resolved_member.linkname)
+
+        # Resulting resolved member may not a regular or hard-linked file, so skip those that case.
+        if resolved_member.isfile() or resolved_member.islnk():
+            f = tar.extractfile(resolved_member)
             file_hash = hash_fileobj(f)
             filename = file_hash[0:HASH_LENGTH] + ".bin"
             to_abs = os.path.join(to_path, filename)
